test/unit/assertions.ts

```diff
@@ -225,6 +225,39 @@ export function testReferenceAssertions(executor: (input: unknown) => void): voi
   })
 }
 
+export function testReferenceOrEnsAssertions(executor: (input: unknown) => void): void {
+  it('should throw exception for bad ReferenceOrEns', async () => {
+    await expect(() => executor(1)).rejects.toThrow(TypeError)
+    await expect(() => executor(true)).rejects.toThrow(TypeError)
+    await expect(() => executor({})).rejects.toThrow(TypeError)
+    await expect(() => executor(null)).rejects.toThrow(TypeError)
+    await expect(() => executor(undefined)).rejects.toThrow(TypeError)
+    await expect(() => executor([])).rejects.toThrow(TypeError)
+    await expect(() => executor('')).rejects.toThrow(TypeError)
+
+    // Not an valid hexstring (ZZZ)
+    await expect(() => executor('ZZZfb5a872396d9693e5c9f9d7233cfa93f395c093371017ff44aa9ae6564cdd')).rejects.toThrow(
+      TypeError,
+    )
+
+    // Prefixed hexstring is not accepted
+    await expect(() => executor('0x634fb5a872396d9693e5c9f9d7233cfa93f395c093371017ff44aa9ae6564cdd')).rejects.toThrow(
+      TypeError,
+    )
+
+    // Length mismatch
+    await expect(() => executor('4fb5a872396d9693e5c9f9d7233cfa93f395c093371017ff44aa9ae6564cdd')).rejects.toThrow(
+      TypeError,
+    )
+
+    // ENS with invalid characters
+    await expect(() => executor('⶿.ß‍.eth')).rejects.toThrow(BeeArgumentError)
```

Comment:
I think we should test these separately
```suggestion
    await expect(() => executor('⶿.eth')).rejects.toThrow(BeeArgumentError)
    await expect(() => executor('ß‍.eth')).rejects.toThrow(BeeArgumentError)
```

Reply:
Hmm I am not sure about this. I took it from the test coverage of the package, so I would just keep it like it: https://github.com/hexonet/idna-uts46/blob/master/test/IdnaTest.txt#L6088

Reply:
Yeah I was assuming this tests non-ASCII chars but it actually tests more than that...